mysite/unmasque/src/core/groupby_clause.py

```diff
@@ -66,6 +67,15 @@ def doExtractJob(self, query):
                 self.see_d_min()
                 new_result = self.app.doJob(query)
 
+                #checking for 1 as constant in select clause.
+                for i in range(len(self.projected_attribs)):
+                    if self.projected_attribs[i] == '' :
+                        for j in range(1, len(new_result)):
+                            if new_result[j][i] != '1':
+                                check_array [i] = 2
```

Comment:
I suppose these 1, 2 values are representative of something, some meaning. Create corresponding variable names as class constants and use them. E.g. CONSTANT_PROJECTION = 1. Then use CONSTANT_PROJECTION wherever 1 is used. This is similar to the use of "enum".

Reply:
Ok ma'am will do it.